### PR TITLE
Add missing set/frozenset methods to ImmutableSet

### LINKED ISSUE #61: Add missing set/frozenset methods to ImmutableSet
- [x] `issubset`
- [x] `issuperset`
- [x] `|` (formatting problem)
- [x] `^` (automatically done once `|` is fixed?)
- [x] `symmetric_difference` (will wrap `^`)
- [x] `copy` (shallow)

### PR DESCRIPTION
**Pull request by berquist**
Closes #61

- [x] Contains a change log entry for Towncrier

### CODE REVIEW

File: immutablecollections/_immutableset.py [hunk 1]
```diff
@@ -210,9 +213,33 @@ def empty() -> "ImmutableSet[T]":
...
 
+    def issubset(self, other: Iterable[T]) -> bool:
+        """
+        This set is a subset of another set if all the elements of this set are
+        are contained in the other set.
+
+        Note that this operation is equivalent to `self <= other`. For determining whether or not
+         this set is a _proper_ subset of another set, use `self < other`.
+        """
+        if isinstance(other, Set):
```
Comment (rgabbard) [inline immutablecollections/_immutableset.py]: What does `isinstance(other, Set)` return if `other` is an `ImmutableSet`?

File: immutablecollections/_immutableset.py [hunk 2]
```diff
@@ -286,6 +312,21 @@ def __rsub__(self, other: AbstractSet[V]) -> "AbstractSet[V]":
...
+            return self ^ other
+        return self ^ immutableset(other)
+
+    def copy(self) -> "ImmutableSet[T]":
+        """
+        Make a shallow copy of this set.
+        """
+        return copy(self)
```
Comment (rgabbard) [inline immutablecollections/_immutableset.py]: Is this added because `set` and `frozenset` have it?  If so, shouldn't is just be `return self` and documented as such?

File: tests/test_immutableset.py [hunk 3]
```diff
@@ -158,41 +159,87 @@ def test_ordering(self):
...
+        self.assertEqual(s1, s3 ^ s2)
+        self.assertEqual(s1, s2.symmetric_difference(s3))
+        self.assertEqual(s1, s3.symmetric_difference(s2))
+
+    def test_copy(self):
+        s1 = immutableset([1, 2, 3, 8])
+        s1c = s1.copy()
+        self.assertEqual(s1, s1c)
+        self.assertEqual(hash(s1), hash(s1c))
+        if python_implementation() == "CPython":
```
Comment (rgabbard) [inline tests/test_immutableset.py]: I don't think we want to enforce this

File: immutablecollections/_immutableset.py (hunk 1 above)
Comment (berquist) [reply immutablecollections/_immutableset.py]: Whose bright idea was it to have `collections.abc.Set` be the same as `typing.AbstractSet` and `collections.abc.MutableSet` be `typing.{Mutable,}Set`...

Comment (berquist) [reply immutablecollections/_immutableset.py]: | (sub)class                 | ImmutableSet is? | `set` | `frozenset` |
|----------------------------|------------------|-------|-------------|
| typing.AbstractSet         | True             | True  | True        |
| typing.MutableSet          | False            | True  | False       |
| typing.Set                 | False            | True  | False       |
| collections.abc.MutableSet | False            | True  | False       |
| collections.abc.Set        | True             | True  | True        |

Comment (rgabbard) [reply immutablecollections/_immutableset.py]: Looks like `typing.AbstractSet` is the way to go.

File: immutablecollections/_immutableset.py [hunk 4]
```diff
@@ -286,6 +310,24 @@ def __rsub__(self, other: AbstractSet[V]) -> "AbstractSet[V]":
...
+        Return the union of elements that are in this set but not the other and elements that are
+        in the other set but not this set (`(self - other) | (other - self)`).
+        """
+        if isinstance(other, AbstractSet):
+            return self ^ other
+        return self ^ immutableset(other)
+
+    def copy(self) -> "ImmutableSet[T]":
+        """
+        For efficiency reasons, rather than provide a copy of this set, return this set directly.
```
Comment (rgabbard) [inline immutablecollections/_immutableset.py]: @berquist : I would reword this even more strongly - something like:

```
Return this set.

Because ImmutableSet is immutable, there is never a reason to copy it. This method is provided for compatibility with `set` and `frozenset`
```

Comment (berquist) [reply immutablecollections/_immutableset.py]: Looks like `frozenset` does the same thing, only `set.copy()` makes a new object.
